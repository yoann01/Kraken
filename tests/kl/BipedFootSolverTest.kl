
require Kraken;

operator entry(){
  BipedFootSolver solver();

  Boolean drawDebug = true;
  Scalar rigScale = 1.0;

  Scalar ikBlend = 1.0;
  Scalar ankleLen =1.67705094814;
  Scalar toeLen = 1.25;
  Mat44 legEnd = Xfo(Quat(Vec3(0.0,0.0,0.0),1.0), Vec3(1.75,1.14999997616,-1.25), Vec3(1.0,1.0,1.0)).toMat44();
  Mat44 ankleIK = Xfo(Quat(Vec3(0.162459865212,-0.688190937042,-0.162459850311),0.688191056252), Vec3(1.75,0.40000000596,0.25), Vec3(1.0,1.0,1.0)).toMat44();
  Mat44 toeIK = Xfo(Quat(Vec3(-0.0,-0.70710682869,0.0),0.70710682869), Vec3(1.75,0.40000000596,0.25), Vec3(1.0,1.0,1.0)).toMat44();
  Mat44 ankleFK = Xfo(Quat(Vec3(0.601501047611,-0.601500928402,0.371748030186),0.37174808979), Vec3(1.75,0.40000000596,0.25), Vec3(1.0,1.0,1.0)).toMat44();
  Mat44 toeFK = Xfo(Quat(Vec3(0.5,-0.5,0.5),0.5), Vec3(1.75,0.40000000596,0.25), Vec3(1.0,1.0,1.0)).toMat44();

  Mat44 ankle_result;
  Mat44 toe_result;

  solver.solve(drawDebug, rigScale, ikBlend, ankleLen, toeLen, legEnd, ankleIK, toeIK, ankleFK, toeFK, ankle_result, toe_result);

  Xfo ankleOutXfo = Xfo(ankle_result);
  Xfo toeOutXfo = Xfo(toe_result);
  report('Ankle Xfo: ' + ankleOutXfo);
  report('Toe Xfo: ' + toeOutXfo);
}
