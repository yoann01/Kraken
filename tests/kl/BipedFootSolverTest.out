[FABRIC:MT] Ignored extension directory 'D:\fabric\FabricEngine-2.1.0-Windows-x86_64\Tests\Exts': unable to open directory 'D:\fabric\FabricEngine-2.1.0-Windows-x86_64\Tests\Exts': The system cannot find the path specified. (code 3 [0x3])Ankle Xfo: {ori:{v:{x:-0.6015,y:+0.601501,z:-0.371748},w:-0.371747},tr:{x:+1.75,y:+1.15,z:-1.25},sc:{x:+1.0,y:+1.0,z:+1.0}}Toe Xfo: {ori:{v:{x:-0.5,y:+0.5,z:-0.5},w:-0.5},tr:{x:+1.749999,y:+0.399999,z:+0.25},sc:{x:+0.999999,y:+0.999999,z:+0.999999}}